### DISCUSSION TITLE
Bricked Wemos D1?

### DISCUSSION THREAD

**Discussion by giovi321** [HASwitchPlate/HASPone#154] [Q&A]
Hi,
I have two HASPone that have been running for more than a year. One of them recently have been acting weird (loss of signal - which was already weak I have to admit - and flashing display intermittently as if the current wasn't stable).
I took it apart, I tried to flash again the Wemos D1 using the Windows utility, I took a look at the pcb and tried to power on the Nextion display separately and it shows the correct page"Initializing". Apparently the problem is that the voltage on the pins that go from the board to the Nextion display have a very unstable voltage ranging from 0 to 5 intermittently. I have tried two different boards and two differend wemos mini D1 - all same behavior.
What could be the cause of this? What else should I try?

PS: forgot to mention that the Wemos D1 mini keeps flashing blue continuously whenever powered (also from pc).

**Comment by giovi321** [comment]
Ok, I found the solution myself!
It's in Italian but you can easily google-translate it:
https://vixr.it/il-wemos-d1-mini-non-funziona-piu-dopo-aver-caricato-un-firmare-ecco-come-ripristinarlo/

**Comment by giovi321** [comment]
Update: whenever I try to configure the HASPone, the Wemos D1 mini bricks again... I don't know if it's the too long password of the wifi (40 chars) but it has always worked fine.

**Reply by giovi321** [reply]
Ok I can confirm that I can reproduce the error with both the wemos mini d1 I own. The password length doesn't matter. Will try with other settings.

**Reply by giovi321** [reply]
The device seems not to retain the mqtt server data when entered. It just bricks. i will try with new devices.

**Comment by aderusha** [comment]
You mention this happens when powered from the PC - is there any chance this is a power supply problem?  Does the problem persist if you use a different USB cable and a standalone charger?  Alternately, do you see the same problem with the device powered by AC?

**Reply by giovi321** [reply]
I tried to power it from both a pc and an apple iphone charger

**Reply by giovi321** [reply]
Oh, I also tried with the original 220v to 5v 3A power supply

**Reply by giovi321** [reply]
I tried a new board just freshly soldered (I thought maybe I did a bad soldering job), but the behavior is still the same.... let's wait for a new wemos d1 mini and see what happens.

**Comment by giovi321** [comment]
Ok, I got a new WEMOS D1 mini, flashed it with the windows flasher and I get the same issue, I can connect to HASPone, confgure it the first time but afther that i get always a white page. I can see the title of the page "HASPone" but nothing else.

**Comment by giovi321** [comment]
I really don't know what to do @aderusha, I have replaced every single component and tried every piece of original code. Maybe you could think of something else that is going wrong or what debugging I could do?

**Reply by giovi321** [reply]
The only difference from the original circuit is that I'm using a c547b transistor that has a different pinout but  I reversed the pinout and it works just fine in the other HASPone

**Comment by aderusha** [comment]
Any chance you have an oscilloscope available?  It'd be interesting to track down the voltage swings you are seeing on the Nextion power rails.  If you don't have stable power there you're going to have a bad time.  The circuit itself is very simple, the NPN transistor is acting as a low-side switch for the Nextion ground.  You should be seeing a stable 5V between GND and +5 on the Nextion connector J2.  Likewise, you should be seeing a stable +3.3V from D6 after the project boots.

![image](https://user-images.githubusercontent.com/9554705/143891276-ced21b28-7175-4203-b61a-f2567a6796a7.png)

**Reply by giovi321** [reply]
There must be a problem somewhere because I see 4.something volts on the Nextion connector and 1.something on the D6 line (on the Wemos). What else could I try?

**Reply by giovi321** [reply]
Great, now it doesn't even create a wifi network anymore... The display just shows a default QR code.
I wonder what is going on with those wemos mini, I mean I did everythis like before and they got bricked somehow? Why am I still able to flash them? Any idea on how to debug them?

**Reply by aderusha** [reply]
> There must be a problem somewhere because I see 4.something volts on the Nextion connector and 1.something on the D6 line (on the Wemos). What else could I try?

If you have a D1 not currently soldered into the PCB, try running the code and check voltage between GND and D6.  This will remove any other variables such as the PCB and components soldered in there.

**Reply by giovi321** [reply]
Running the code with arduino you mean?

**Reply by giovi321** [reply]
By just plugging the D1 into a usb port of my laptop the voltage between GND and D6 is 3.27v (measured with a cheap mutimeter).

**Reply by giovi321** [reply]
I'm not a master at soldering but neither a newbie. Could it be some fux shortting come pins? Maybe I should dip it into IPA.
In any case that would be a very weird option: 3/3 boards failling?

**Reply by aderusha** [reply]
3.27v is just what we would expect, so from that we know that the D1 is applying the correct voltage to that low-side switch circuit. It'd be interesting to be able to track that voltage over time, and compare against the voltage seen on the J2 connector.  Do you have a scope available to you?

**Reply by giovi321** [reply]
Unfortunately all I have is a multimeter... No scope at all

**Reply by giovi321** [reply]
On wednesday I'm receiving a new wemos. I'll start from scratch: new board, new assembly, new nextion.

**Comment by giovi321** [comment]
Tried again with a new board, new wemos, new nextion display but I experience pretty much the same: I can set up the first time the HASPone but then when I try to save the settings through the web interface it stops working and disconnects. 
Never successfully connected to the mqtt broker even if correctly set.

**Comment by aderusha** [comment]
Does any of this happen on older versions of the firmware?  You can find a list of URLs you can use in the FW update web page here: https://github.com/HASwitchPlate/HASPone/tree/main/update

**Reply by giovi321** [reply]
No success with the older versions

**Comment by aderusha** [comment]
Also - can you capture debug output on the boot cycle and paste it here?  That might tell us something about what is going wrong, here's how to do that: https://github.com/HASwitchPlate/HASPone/wiki/Capture-HASPone-debug-output

**Reply by giovi321** [reply]
Here's the USB serial output, I cut it to few lines and redacted passwords etc.
https://gist.github.com/giovi321/ed03f97471710d82d0669aacc02ad6e0
Edit: obviously the wemos is disconnected from anything else here, is this what you expected?

**Reply by giovi321** [reply]
After that, the device is bricked and needs to be flashed with a blank, then flashed again with HASPone

**Reply by aderusha** [reply]
So if you take the devices which have been running for a year, flash them back to the version they were running previously, do they still crash at boot time?

**Reply by giovi321** [reply]
Well, the one that is still running i won't even touch it with a feather :laughing: 
The other one that has been touched (may I be damned) follows this process:

1. Flash first time WEMOS and upload tft file to the Nextion display
2. Insert device into the board (newly soldered) and connect new Nextion display
3. Apply power to the board with a high quality 3A 5v power supply
4. Connect to HASPone with my iPhone and set the wifi credentials
5. Device reboots
6. Connect to the HASPone with my laptop and set the remaining settings (such as mqtt) and save the settings
7. Device doesn't reboot nor connects to the wifi network to which was connected before nor it creates a new netword (it just spits out what I pasted before)
8. OPTIONAL: flash blank on the device and restart from 1

This is pretty much what I have been doing for the last 4 days or so

EDIT: maybe it could be useful to do a USB debug at a different step?

**Reply by giovi321** [reply]
Here is a more complete output from the very first boot of the device.
https://gist.github.com/giovi321/2cc35823d0280e4d24389234ec67fede
I mean, what could I be doing wrong with the hardware?? It's so simple!

**Reply by aderusha** [reply]
That log is super helpful and I notice something going amiss with the Nextion serial speed value.  This would also align with the crash happening immediately after reading `config.json` as the very next step in the boot process is to start talking to the Nextion.

Can you try updating your device using this URL? `https://raw.githubusercontent.com/HASwitchPlate/HASPone/dev-1.05/Arduino_Sketch/HASwitchPlate.ino.d1_mini.bin`

That's a testing version of the next HASPone firmware which has some logic to check for missing Nextion serial communication details and which selects a sensible default if they weren't loaded from `config.json` for whatever reason.

**Reply by giovi321** [reply]
Ok, we went one step forward. Now the wemos does not brick but it keeps connecting and disconnecting from the mqtt server as per logs attached:
https://gist.github.com/giovi321/71e6ae1f6c7ed5c6e39a4ed6488b0742

**Reply by aderusha** [reply]
Hopefully it will Just Work and you’ll be all set :D. But if not… yeah those debug logs are really handy, so post the result if it’s still crashing and we’ll keep working this until we got you sorted!

**Reply by giovi321** [reply]
See previous post :)

**Reply by giovi321** [reply]
Veeeery weird behaviour I'm seeing here. I just set wrong mqtt credentials, then correct ones and it connected. However couldn't see any page. More logs coming.

**Reply by giovi321** [reply]
Ok, most recent evolution of the events. I can see one icon on the nextion display but not interact with anything.
This is the serial output:
https://gist.github.com/giovi321/643f79a8b9afc2b80353f6ee8a59dc2c

**Reply by aderusha** [reply]
Allright, a few things I'm seeing here:

Can you confirm that you are using [this Nextion firmware](https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate.tft) (or more to the point, can you try flashing with that specific file)?  I ask because of the following lines in the boot process:

```
[+0.627s] HMI: Querying LCD firmware version number
[+0.634s] HMI OUT: get p[0].b[2].val
[+0.637s] HMI IN: [4 bytes]: 0x1a 0xff 0xff 0xff
[+0.642s] HMI IN: [Invalid Variable name or attribute] 0x1a
[+2.230s] HMI: LCD version query timed out
```

That command should have returned a value and it suggests that there might be something wrong with the code currently running on the Nextion display.

Additionally, can you try deploying the Core Functionality blueprint but with no non-default options selected apart from the device?  So, disable your existing Core Functionality automation, deploy a new one temporarily, select your HASPone, leave the rest of the defaults, Save, Run Actions.  Does the device work?

**Reply by giovi321** [reply]
Ok I understood how blueprints work - very smart. Even if I created and ran the automation, the display just says connected to wifi and to mqtt, doesn't show the fist page if not for a second. I feel we are getting closer to the solution (except that my nextion should be horizontal :laughing: )
(BTW, I got you some coffee).

**Reply by aderusha** [reply]
After you run the Core Functionality blueprint, it should throw a notification with the YAML code you need.  [See this video](https://www.youtube.com/watch?v=hTb49n4butI&t=531s) for a look at how that works.

How is the rest of it going?

**Reply by giovi321** [reply]
Ok, now showing pages, although evey time I tap in a page number it shows the "wifi connected mwtt connected" screen for half of a second.
EDIT: check also my previous comment, sorry I keep editing them.

**Reply by aderusha** [reply]
Can you dump a log of the current situation?  WE'RE GETTING SO CLOSE!

**Reply by aderusha** [reply]
Also thanks for that coffee :D

**Reply by giovi321** [reply]
Here's the gist!
https://gist.github.com/giovi321/c1da21be132a993f20f53e57fe288d97
I also tried to scroll through the pages a few times

**Reply by giovi321** [reply]
We are so close! What's the next step?

**Reply by giovi321** [reply]
@aderusha any suggestion :pleading_face:

**Reply by aderusha** [reply]
Taking a look at that log above... I'm not seeing any errors nor sign of WiFi or MQTT dropping connections.  The trace shows the device starting up, configuration being sent via blueprints, and the user cycling through the 3 pages using the page control buttons.

Did something unexpected happen while grabbing that log?

**Reply by giovi321** [reply]
Nothing weird happened, except that every time you switch the page you get the reconnecting screen.
Is there a way to excract a .bin file from the wemos (the working one) and flash it to the new one? I googled it a bit but didn't really find an answer.

**Reply by aderusha** [reply]
Duplicating the flash on an ESP8266 isn't impossible but it's not straightforward and probably isn't the solution here.  I just saw something in the logs that is a little unexpected - something outside of the HASPone is sending it that `page 0` command with every button press, which strongly suggests a config problem on the Home Assistant side.  Specifically, some automation on Home Assistant is telling your device to switch to page 0, and then to flip to the desired page.  Can you [run through this](https://github.com/HASwitchPlate/HASPone/wiki/Fix-common-HASPone-issues) and confirm that you only have one HASPone device with that device name, and no other core Functionality blueprints have been deployed for this device?

**Reply by giovi321** [reply]
Yes I can confirm I have only one HaspONE (I have two other haswitchplates but are old and have totally different names from plate01).
Now it's interesting, when scrolling through the pages I don't get to see page 0 anymore but I see a weird effect as if It went through many pages very quickly.

Here's a video of it https://user-images.githubusercontent.com/6443515/144638372-21571bbc-be57-4af1-81c2-8b81441fadaa.MOV

**Reply by giovi321** [reply]
Ok I found out why, some automations were triplicated. Where are those stored physically?

**Reply by giovi321** [reply]
I think we can close this massive topic, I learned  a lot. Thank you very much @aderusha.
